test/controllers/actions/AuthActionSpec.scala

```diff
@@ -17,983 +17,482 @@
-      "must fail with a UNAUTHORIZED when there is no AGENT REFERENCE NUMBER enrolment" in {
-
-        val application = applicationBuilder(userAnswers = None).build()
-
-        val enrolments: Enrolments = Enrolments(Set())
-
-        val authResponse: Option[AffinityGroup] ~ Enrolments ~ ConfidenceLevel =
-          new ~(new ~(
-            Some(AffinityGroup.Agent),
-            enrolments),
-            ConfidenceLevel.L50)
+      "should return expected result for Agent happy path" in new Test {
+        val partialAuthResponse: Option[AffinityGroup] ~ Enrolments = new ~(
+          Some(AffinityGroup.Agent),
+          Enrolments(Set(HMRCEnrolment(
+            key = Enrolment.Agent.key,
+            identifiers = Seq(EnrolmentIdentifier(Enrolment.Agent.value, "value")),
+            state = "activated"
+          )))
+        )
 
-        running(application) {
+        val authResponse: Option[AffinityGroup] ~ Enrolments ~ ConfidenceLevel = new ~(
+          partialAuthResponse,
+          ConfidenceLevel.L250
+        )
 
-          val bodyParsers = application.injector.instanceOf[BodyParsers.Default]
-          val appConfig = application.injector.instanceOf[FrontendAppConfig]
+        when(mockAuthConnector.authorise[AuthReturnType](
+          predicate = any(),
+          retrieval = ArgumentMatchers.eq(affinityGroup and allEnrolments and confidenceLevel)
+        )(any(), any()))
+          .thenReturn(Future.successful(authResponse))
 
-          val authAction = new IdentifierActionProviderImpl(new FakeSuccessfulAuthConnector(authResponse), appConfig,
-            mockSessionDataConnector, bodyParsers)(ec).apply(taxYear)
-          val controller = new Harness(authAction)
+        when(mockAuthConnector.authorise[Unit](
+          predicate = any(),
+          retrieval = ArgumentMatchers.eq(EmptyRetrieval)
+        )(any(), any()))
+          .thenReturn(Future.successful())
 
-          when(mockSessionDataConnector.getSessionData(any())).thenReturn(
-            Future.successful(Right(Some(SessionData("1234567890", "nino", "utr", "test-session-id"))))
-          )
+        mockSessionServiceEnabled(true)
+        mockGetSessionDataFromSessionStore(Right(Some(dummySessionData)))
 
-          val result = controller.onPageLoad()(FakeRequest().withSession("ClientMTDID" -> "1234567890", "sessionId" -> "test-session-id"))
+        val result: Future[Result] = testAuth.invokeBlock(
+          request = FakeRequest().withHeaders(("sessionId", sessionId)),
+          block = (_: IdentifierRequest[_]) => Future.successful(ImATeapot("Teapot time"))
+        )
 
-          status(result) mustBe SEE_OTHER
-          redirectLocation(result) mustBe Some("https://www.gov.uk/guidance/get-an-hmrc-agent-services-account")
-        }
+        status(result) mustBe IM_A_TEAPOT
+        contentAsString(result) mustBe "Teapot time"
       }
     }
 
-    "when the user is authorised as a secondary agent" - {
+    "nonAgentAuthentication" - {
```

Comment:
These are testing private functions, this behaviour should instead be tested via the invokeBlock method